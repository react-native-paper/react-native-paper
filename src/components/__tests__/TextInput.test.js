import * as React from 'react';
import { render } from 'react-native-testing-library';
import TextInput from '../TextInput/TextInput';

const affixTextValue = '/100';
it('correctly renders left-side icon adornment, and right-side affix adornment', () => {
  const { getByText, getByTestId, toJSON } = render(
    <TextInput
      label="Flat input"
      placeholder="Type something"
      value={'Some test value'}
      onChangeText={(text) => this.setState({ text })}
      left={
        <TextInput.Icon
          name="heart"
          onPress={() => {
            console.log('!@# press left');
          }}
        />
      }
      right={
        <TextInput.Affix
          text={affixTextValue}
          textStyle={{ color: '#FF0000' }}
        />
      }
    />
  );
  expect(() => getByText(affixTextValue)).not.toThrow();
  expect(() => getByTestId('left-icon-adornment')).not.toThrow();
  expect(() => getByTestId('right-affix-adornment')).not.toThrow();
  expect(toJSON()).toMatchSnapshot();
});

it('correctly renders left-side icon adornment, and right-side affix adornment ', () => {
  const { getByText, getByTestId, toJSON } = render(
    <TextInput
      label="Flat input"
      placeholder="Type something"
      value={'Some test value'}
      onChangeText={(text) => this.setState({ text })}
      left={
        <TextInput.Affix
          text={affixTextValue}
          textStyle={{ color: '#FF0000' }}
        />
      }
      right={
        <TextInput.Icon
          name="heart"
          onPress={() => {
            console.log('!@# press left');
          }}
        />
      }
    />
  );
  expect(() => getByText(affixTextValue)).not.toThrow();
  expect(() => getByTestId('right-icon-adornment')).not.toThrow();
  expect(() => getByTestId('left-affix-adornment')).not.toThrow();
  expect(toJSON()).toMatchSnapshot();
});

it('correctly applies default textAlign based on default RTL', () => {
  const { toJSON } = render(
    <TextInput
      label="Flat input"
      placeholder="Type something"
      value={'Some test value'}
    />
  );

  expect(toJSON()).toMatchSnapshot();
});

it('correctly applies textAlign center', () => {
  const { toJSON } = render(
    <TextInput
      label="Flat input"
      placeholder="Type something"
      value={'Some test value'}
      style={{ textAlign: 'center' }}
    />
  );

  expect(toJSON()).toMatchSnapshot();
});

it('uses paddingTop to position minimized label', () => {
  const { toJSON } = render(
    <TextInput
      label="Flat input"
      placeholder="Type something"
      value={'Some test value'}
      dense
      style={{
        height: 40,
        paddingTop: 4,
        paddingHorizontal: 4,
      }}
    />
  );

  expect(toJSON()).toMatchSnapshot();
});
