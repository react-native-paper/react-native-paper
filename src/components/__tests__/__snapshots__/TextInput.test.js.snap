// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correctly applies default textAlign based on default RTL 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgb(231, 231, 231)",
        "borderTopLeftRadius": 4,
        "borderTopRightRadius": 4,
      },
      Object {},
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.26)",
        "bottom": 0,
        "height": 2,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "transform": Array [
          Object {
            "scaleY": 0.5,
          },
        ],
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "paddingBottom": 0,
          "paddingTop": 0,
        },
        Object {
          "minHeight": 64,
        },
      ]
    }
  >
    <View
      pointerEvents="none"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": Array [
            Object {
              "translateX": 3,
            },
          ],
        }
      }
    >
      <Text
        numberOfLines={1}
        onLayout={[Function]}
        style={
          Object {
            "color": "#6200ee",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": undefined,
            "left": 0,
            "lineHeight": undefined,
            "opacity": 0,
            "paddingLeft": 12,
            "paddingRight": 12,
            "position": "absolute",
            "textAlign": "left",
            "top": 34,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": -16,
              },
              Object {
                "scale": 0.75,
              },
            ],
            "writingDirection": "ltr",
          }
        }
      >
        Flat input
      </Text>
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": undefined,
            "left": 0,
            "lineHeight": undefined,
            "opacity": 0,
            "paddingLeft": 12,
            "paddingRight": 12,
            "position": "absolute",
            "textAlign": "left",
            "top": 34,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": -16,
              },
              Object {
                "scale": 0.75,
              },
            ],
            "writingDirection": "ltr",
          }
        }
      >
        Flat input
      </Text>
    </View>
    <TextInput
      allowFontScaling={true}
      editable={true}
      multiline={false}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder=""
      placeholderTextColor="rgba(0, 0, 0, 0.54)"
      rejectResponderTermination={true}
      selectionColor="#6200ee"
      style={
        Array [
          Object {
            "flexGrow": 1,
            "margin": 0,
            "zIndex": 1,
          },
          Object {
            "paddingLeft": 12,
            "paddingRight": 12,
          },
          Object {
            "height": 64,
          },
          Object {
            "paddingBottom": 4,
            "paddingTop": 24,
          },
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": undefined,
            "textAlign": "left",
            "textAlignVertical": "center",
          },
          false,
          Array [
            Object {},
          ],
        ]
      }
      underlineColorAndroid="transparent"
      value="Some test value"
    />
  </View>
</View>
`;

exports[`correctly applies height to multiline Outline TextInput 1`] = `
<View
  style={Object {}}
>
  <View>
    <View
      pointerEvents="none"
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 6,
          },
          Object {
            "backgroundColor": "#f6f6f6",
            "borderColor": "rgba(0, 0, 0, 0.54)",
            "borderRadius": 4,
            "borderWidth": 1,
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "paddingBottom": 0,
          },
          Object {
            "minHeight": 100,
            "paddingTop": 8,
          },
        ]
      }
    >
      <View
        pointerEvents="none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 3,
              },
            ],
          }
        }
      >
        <View
          pointerEvents="none"
          style={
            Object {
              "backgroundColor": "#f6f6f6",
              "bottom": 4,
              "left": 10,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 6,
              "transform": Array [
                Object {
                  "translateX": -3,
                },
              ],
              "width": 8,
            }
          }
        />
        <Text
          numberOfLines={1}
          style={
            Object {
              "backgroundColor": "#f6f6f6",
              "color": "transparent",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "left": 18,
              "opacity": 1,
              "paddingHorizontal": 0,
              "position": "absolute",
              "textAlign": "left",
              "top": 59,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": -52,
                },
                Object {
                  "scale": 0.75,
                },
                Object {
                  "scaleY": 0.2,
                },
              ],
              "writingDirection": "ltr",
            }
          }
        >
          Outline Input
        </Text>
        <Text
          numberOfLines={1}
          onLayout={[Function]}
          style={
            Object {
              "color": "#6200ee",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "left": 0,
              "opacity": 0,
              "paddingHorizontal": 14,
              "position": "absolute",
              "textAlign": "left",
              "top": 58,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": -52,
                },
                Object {
                  "scale": 0.75,
                },
              ],
              "writingDirection": "ltr",
            }
          }
        >
          Outline Input
        </Text>
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.54)",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "left": 0,
              "opacity": 0,
              "paddingHorizontal": 14,
              "position": "absolute",
              "textAlign": "left",
              "top": 58,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": -52,
                },
                Object {
                  "scale": 0.75,
                },
              ],
              "writingDirection": "ltr",
            }
          }
        >
          Outline Input
        </Text>
      </View>
      <TextInput
        allowFontScaling={true}
        editable={true}
        multiline={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder=""
        placeholderTextColor="rgba(0, 0, 0, 0.54)"
        rejectResponderTermination={true}
        selectionColor="#6200ee"
        style={
          Array [
            Object {
              "flexGrow": 1,
              "margin": 0,
              "paddingHorizontal": 14,
              "zIndex": 1,
            },
            Object {
              "height": 100,
            },
            Object {
              "paddingBottom": 24,
              "paddingTop": 24,
            },
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "textAlign": "left",
              "textAlignVertical": "top",
            },
            false,
            Array [
              Object {},
            ],
          ]
        }
        underlineColorAndroid="transparent"
        value="Some test value"
      />
    </View>
  </View>
</View>
`;

exports[`correctly applies textAlign center 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgb(231, 231, 231)",
        "borderTopLeftRadius": 4,
        "borderTopRightRadius": 4,
      },
      Object {},
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.26)",
        "bottom": 0,
        "height": 2,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "transform": Array [
          Object {
            "scaleY": 0.5,
          },
        ],
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "paddingBottom": 0,
          "paddingTop": 0,
        },
        Object {
          "minHeight": 64,
        },
      ]
    }
  >
    <View
      pointerEvents="none"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": Array [
            Object {
              "translateX": 3,
            },
          ],
        }
      }
    >
      <Text
        numberOfLines={1}
        onLayout={[Function]}
        style={
          Object {
            "color": "#6200ee",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": undefined,
            "left": 0,
            "lineHeight": undefined,
            "opacity": 0,
            "paddingLeft": 12,
            "paddingRight": 12,
            "position": "absolute",
            "textAlign": "left",
            "top": 34,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": -16,
              },
              Object {
                "scale": 0.75,
              },
            ],
            "writingDirection": "ltr",
          }
        }
      >
        Flat input
      </Text>
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": undefined,
            "left": 0,
            "lineHeight": undefined,
            "opacity": 0,
            "paddingLeft": 12,
            "paddingRight": 12,
            "position": "absolute",
            "textAlign": "left",
            "top": 34,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": -16,
              },
              Object {
                "scale": 0.75,
              },
            ],
            "writingDirection": "ltr",
          }
        }
      >
        Flat input
      </Text>
    </View>
    <TextInput
      allowFontScaling={true}
      editable={true}
      multiline={false}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder=""
      placeholderTextColor="rgba(0, 0, 0, 0.54)"
      rejectResponderTermination={true}
      selectionColor="#6200ee"
      style={
        Array [
          Object {
            "flexGrow": 1,
            "margin": 0,
            "zIndex": 1,
          },
          Object {
            "paddingLeft": 12,
            "paddingRight": 12,
          },
          Object {
            "height": 64,
          },
          Object {
            "paddingBottom": 4,
            "paddingTop": 24,
          },
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": undefined,
            "textAlign": "center",
            "textAlignVertical": "center",
          },
          false,
          Array [
            Object {},
          ],
        ]
      }
      underlineColorAndroid="transparent"
      value="Some test value"
    />
  </View>
</View>
`;

exports[`correctly renders left-side icon adornment, and right-side affix adornment  1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgb(231, 231, 231)",
        "borderTopLeftRadius": 4,
        "borderTopRightRadius": 4,
      },
      Object {},
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.26)",
        "bottom": 0,
        "height": 2,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "transform": Array [
          Object {
            "scaleY": 0.5,
          },
        ],
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "paddingBottom": 0,
          "paddingTop": 0,
        },
        Object {
          "minHeight": 64,
        },
      ]
    }
  >
    <View
      pointerEvents="none"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": Array [
            Object {
              "translateX": 3,
            },
          ],
        }
      }
    >
      <Text
        numberOfLines={1}
        onLayout={[Function]}
        style={
          Object {
            "color": "#6200ee",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": undefined,
            "left": 0,
            "lineHeight": undefined,
            "opacity": 0,
            "paddingLeft": 12,
            "paddingRight": 44,
            "position": "absolute",
            "textAlign": "left",
            "top": 34,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": -16,
              },
              Object {
                "scale": 0.75,
              },
            ],
            "writingDirection": "ltr",
          }
        }
      >
        Flat input
      </Text>
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": undefined,
            "left": 0,
            "lineHeight": undefined,
            "opacity": 0,
            "paddingLeft": 12,
            "paddingRight": 44,
            "position": "absolute",
            "textAlign": "left",
            "top": 34,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": -16,
              },
              Object {
                "scale": 0.75,
              },
            ],
            "writingDirection": "ltr",
          }
        }
      >
        Flat input
      </Text>
    </View>
    <TextInput
      allowFontScaling={true}
      editable={true}
      multiline={false}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder=""
      placeholderTextColor="rgba(0, 0, 0, 0.54)"
      rejectResponderTermination={true}
      selectionColor="#6200ee"
      style={
        Array [
          Object {
            "flexGrow": 1,
            "margin": 0,
            "zIndex": 1,
          },
          Object {
            "paddingLeft": 12,
            "paddingRight": 44,
          },
          Object {
            "height": 64,
          },
          Object {
            "paddingBottom": 4,
            "paddingTop": 24,
          },
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": undefined,
            "textAlign": "left",
            "textAlignVertical": "center",
          },
          false,
          Object {
            "marginLeft": 0,
            "marginRight": 36,
            "paddingLeft": 36,
            "paddingRight": 8,
          },
        ]
      }
      underlineColorAndroid="transparent"
      value="Some test value"
    />
  </View>
  <View
    onLayout={[Function]}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "left": 12,
        "opacity": 1,
        "position": "absolute",
        "top": null,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
          },
          Object {
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": undefined,
          },
          Object {
            "color": "#FF0000",
          },
        ]
      }
    >
      /100
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "height": 24,
          "justifyContent": "center",
          "position": "absolute",
          "width": 24,
        },
        Object {
          "right": 12,
          "top": 20,
        },
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": undefined,
        }
      }
      accessible={true}
      focusable={true}
      hitSlop={
        Object {
          "bottom": 6,
          "left": 6,
          "right": 6,
          "top": 6,
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "margin": 6,
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 18,
              "height": 36,
              "width": 36,
            },
            undefined,
            Object {
              "margin": 0,
            },
          ],
        ]
      }
    >
      <View>
        <Text
          accessibilityElementsHidden={true}
          allowFontScaling={false}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          selectable={false}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontSize": 24,
              },
              Array [
                Object {
                  "lineHeight": 24,
                  "transform": Array [
                    Object {
                      "scaleX": 1,
                    },
                  ],
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ],
              Object {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`correctly renders left-side icon adornment, and right-side affix adornment 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgb(231, 231, 231)",
        "borderTopLeftRadius": 4,
        "borderTopRightRadius": 4,
      },
      Object {},
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.26)",
        "bottom": 0,
        "height": 2,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "transform": Array [
          Object {
            "scaleY": 0.5,
          },
        ],
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "paddingBottom": 0,
          "paddingTop": 0,
        },
        Object {
          "minHeight": 64,
        },
      ]
    }
  >
    <View
      pointerEvents="none"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": Array [
            Object {
              "translateX": 11,
            },
          ],
        }
      }
    >
      <Text
        numberOfLines={1}
        onLayout={[Function]}
        style={
          Object {
            "color": "#6200ee",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": undefined,
            "left": 0,
            "lineHeight": undefined,
            "opacity": 0,
            "paddingLeft": 44,
            "paddingRight": 44,
            "position": "absolute",
            "textAlign": "left",
            "top": 34,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": -16,
              },
              Object {
                "scale": 0.75,
              },
            ],
            "writingDirection": "ltr",
          }
        }
      >
        Flat input
      </Text>
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": undefined,
            "left": 0,
            "lineHeight": undefined,
            "opacity": 0,
            "paddingLeft": 44,
            "paddingRight": 44,
            "position": "absolute",
            "textAlign": "left",
            "top": 34,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": -16,
              },
              Object {
                "scale": 0.75,
              },
            ],
            "writingDirection": "ltr",
          }
        }
      >
        Flat input
      </Text>
    </View>
    <TextInput
      allowFontScaling={true}
      editable={true}
      multiline={false}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder=""
      placeholderTextColor="rgba(0, 0, 0, 0.54)"
      rejectResponderTermination={true}
      selectionColor="#6200ee"
      style={
        Array [
          Object {
            "flexGrow": 1,
            "margin": 0,
            "zIndex": 1,
          },
          Object {
            "paddingLeft": 44,
            "paddingRight": 44,
          },
          Object {
            "height": 64,
          },
          Object {
            "paddingBottom": 4,
            "paddingTop": 24,
          },
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": undefined,
            "textAlign": "left",
            "textAlignVertical": "center",
          },
          false,
          Object {
            "marginLeft": 36,
            "marginRight": 0,
            "paddingLeft": 8,
            "paddingRight": 36,
          },
        ]
      }
      underlineColorAndroid="transparent"
      value="Some test value"
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "height": 24,
          "justifyContent": "center",
          "position": "absolute",
          "width": 24,
        },
        Object {
          "left": 12,
          "top": 20,
        },
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": undefined,
        }
      }
      accessible={true}
      focusable={true}
      hitSlop={
        Object {
          "bottom": 6,
          "left": 6,
          "right": 6,
          "top": 6,
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "margin": 6,
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 18,
              "height": 36,
              "width": 36,
            },
            undefined,
            Object {
              "margin": 0,
            },
          ],
        ]
      }
    >
      <View>
        <Text
          accessibilityElementsHidden={true}
          allowFontScaling={false}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          selectable={false}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontSize": 24,
              },
              Array [
                Object {
                  "lineHeight": 24,
                  "transform": Array [
                    Object {
                      "scaleX": 1,
                    },
                  ],
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ],
              Object {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
  <View
    onLayout={[Function]}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "right": 12,
        "top": null,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
          },
          Object {
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": undefined,
          },
          Object {
            "color": "#FF0000",
          },
        ]
      }
    >
      /100
    </Text>
  </View>
</View>
`;
